clang/lib/CIR/CodeGen/CIRGenExpr.cpp

```diff
@@ -224,6 +224,10 @@ void CIRGenFunction::emitStoreThroughLValue(RValue src, LValue dst,
       return;
     }
 
+    assert(dst.isBitField() && "NIY LValue type");
```

Comment:
```suggestion
    assert(dst.isBitField() && "NYI LValue type");
```

Also, why is this an assert here for an NYI?  This should use the errorNYI stuff instead, perhaps just put htis in a `isBitField` branch?

Reply:
I think this is an assert in the classic codegen path. If we reach this point and it's not a bitfield, it says it's an "unknown Lvalue type":
https://github.com/llvm/llvm-project/blob/ff365ce1d71df6231a4b9d37f8d4bfc6d35a2a34/clang/lib/CodeGen/CGExpr.cpp#L2526

So, instead of treating it as a NYI case, I think it's better to just update the message, since this isn’t actually an unimplemented feature.